openmdao/docs/_utils/docutil.py

```diff
@@ -383,52 +399,63 @@ def insert_output_start_stop_indicators(src):
+                       'print(',
+                       '.run_model(',
+                       '.run_driver(',
+                       '.check_partials(',
+                       '.check_totals(',
+                       '.list_inputs(',
+                       '.list_outputs(',
+                       ]
+
+    newlines = []
     input_block_number = 0
-
-    # find all nodes that might produce output
-    nodes = rb.findAll(lambda identifier: identifier in ['print', 'atomtrailers'])
-    for r in nodes:
-        # assume that whatever is in the try block will fail and produce no output
-        # this way we can properly handle display of error messages in the except
-        if hasattr(r.parent, 'type') and r.parent.type == 'try':
+    in_try = False
+    in_continuation = False
+    head_indent = ''
+    for line in lines:
+        newlines.append(line)
+
+        # Check if we are concluding a continuation line.
+        if in_continuation:
+            line = line.rstrip()
+            if not (line.endswith(',') or line.endswith('\\') or line.endswith('(')):
+                newlines.append('%sprint(">>>>>%d")' % (head_indent, input_block_number))
+                input_block_number += 1
+                in_continuation = False
+
+        # Don't print if we are in a try block.
+        if in_try:
+            if 'except' in line:
+                in_try = False
             continue
 
-        # Output within if/else statements is not a good idea for docs, because
-        # we don't know which branch execution will follow and thus where to put
-        # the output block. Regardless of which branch is taken, though, the
-        # output blocks must start with the same block number.
-        if hasattr(r.parent, 'type') and r.parent.type == 'if':
-            if_block_number = input_block_number
-        if hasattr(r.parent, 'type') and r.parent.type in ['elif', 'else']:
-            input_block_number = if_block_number
+        if 'try:' in line:
+            in_try = True
+            continue
 
-        if is_output_node(r):
-            # if there was a trailing comment on this line, output goes after it
-            if r in lines_with_comments:
-                r = lines_with_comments[r]  # r is now the comment
+        # Searching for 'print(' is a little ambiguous.
+        if 'set_solver_print(' in line:
+            continue
 
-            # find the correct node to 'insert_after'
-            while hasattr(r, 'parent') and not hasattr(r.parent, 'insert'):
-                r = r.parent
+        for item in print_producing:
+            if item in line:
+                indent = '' * (len(line) - len(line.lstrip()))
 
-            r.insert_after('print(">>>>>%d")\n' % input_block_number)
-            input_block_number += 1
+                # Line continuations are a litle tricky.
+                line = line.rstrip()
+                if line.endswith(',') or line.endswith('\\') or line.endswith('('):
```

Comment:
This won't catch a number of possible continuations.  You'll probably need to keep track of the number of open and closed parens, square brackets, and curly braces as well to make this more robust.

Reply:
We don't need to put any of that stuff inside of function calls. Let's write nice readable tests for the poor user.